copilot/ask_view.py

```diff
@@ -0,0 +1,295 @@
+<div style="padding-right:2px">
+	<a style="{PHANTOM_A_TAG_STYLE}" href ="save_{{id}}">{{save}}</a>
+	<a style="{PHANTOM_A_TAG_STYLE}" href="copy_{{id}}">{{copy}}</a>
+</div>
+"""
+
+class CopilotViewManager:
+	can_type = True
+
+	@property
+	def gpt_view(self) -> sublime.View:
+		if not getattr(CopilotViewManager, "_gpt_view",None):
+			# File config and creation
+			CopilotViewManager._gpt_view = sublime.active_window().new_file(syntax="Packages/Markdown/Markdown.sublime-syntax")	
+			CopilotViewManager.can_type = True
+			CopilotViewManager._gpt_view.settings().set("PIECES_GPT_VIEW",True) # Label the view as gpt view
+			CopilotViewManager._gpt_view.settings().set("end_response",0) # End reponse charater
+			CopilotViewManager._gpt_view.settings().set("line_numbers", False) # Remove lines
+			CopilotViewManager._gpt_view.set_scratch(True)
+			CopilotViewManager._gpt_view.set_name(self.view_name)
+
+			# Phantom intilization 
+			self.last_edit_phantom = 0
+			self.phantom_set = sublime.PhantomSet(CopilotViewManager._gpt_view, "Pieces_Phantoms")
+			self.phantom_details_dict = {} # id: {"code":code,"region":region}
+
+
+
+			# Failed regions
+			self.failed_regions = []
+			self.failed_phantom = sublime.PhantomSet(CopilotViewManager._gpt_view, "Pieces_Failed_Phantoms")
+
+			# Others
+			self.copilot_regions = []
+			self.show_cursor
+			self.update_status_bar()
+			self.render_copilot_image_phantom(CopilotViewManager._gpt_view)
+
+        
+			# Create a new group (split view)
+			sublime.active_window().run_command("set_layout", {
+			    "cols": [0.0, 0.5, 1.0],
+			    "rows": [0.0, 1.0],
+			    "cells": [[0, 0, 1, 1], [1, 0, 2, 1]]
+			})
+
+			# Move the active view to the new group
+			sublime.active_window().set_view_index(CopilotViewManager._gpt_view, 1, 0)
+
+			# Focus on the new group
+			sublime.active_window().focus_group(1)
+		return CopilotViewManager._gpt_view
+		
+	@property
+	def view_name(self):
+		return "Pieces: " + getattr(self,"_view_name","New Conversation")
+	
+	@view_name.setter
+	def view_name(self,v):
+		self._view_name = v
+		self.gpt_view.set_name(self.view_name)
+	
+
+	@gpt_view.setter
+	def gpt_view(self,view):
+		CopilotViewManager._gpt_view = view
+
+
+	def update_status_bar(self):
+		if getattr(self,"_gpt_view",None):
+			self._gpt_view.set_status("MODEL",f"LLM Model: {PiecesSettings.model_name.replace('Chat Model','')}")
+
+	@property
+	def show_cursor(self):
+		self.gpt_view.set_status("MODEL",PiecesSettings.model_name)
+		self.gpt_view.run_command("append",{"characters":">>> "})
+		self.end_response = self.end_response + 4  # ">>> " 4 characters
+		ui = sublime.ui_info()["theme"]["style"]
+
+		self.copilot_regions.append(sublime.Region(self.gpt_view.size(), self.gpt_view.size()))
+
+		# Add the regions with the icon and appropriate flags
+		self.gpt_view.add_regions(
+			"pieces", 
+			self.copilot_regions, 
+			scope="text", 
+			icon=f"Packages/Pieces/copilot/images/copilot-icon-{ui}.png", 
```

Comment:
this works on windows?

Reply:
yes no worries these paths are handled by sublime